src/lib/util/Config.ts

```diff
@@ -128,25 +128,40 @@ export class Config {
 		// If the config specifies property, use those.
 		if (ecc && ecc[propertyName] && typeChecker(ecc[propertyName], propertyName, engine)) {
 			this.logger.trace(`Retrieving ${propertyName} from engine ${engine}: ${ecc[propertyName]}`);
-			return ecc[propertyName];
 		} else {
-			// If the config doesn't specify the property, see if we have any default values.
-			const defaultConfig = DEFAULT_CONFIG.engines.find(e => e.name.toLowerCase() === engine.valueOf().toLowerCase());
-			// If we find default values, persist those to the config.
-			if (!ecc) {
-				ecc = defaultConfig;
-				this.logger.warn(`Persisting missing block for engine ${engine}`);
-				await this.writeConfig();
-			} else if (defaultConfig[propertyName]) {
-				ecc[propertyName] = defaultConfig[propertyName];
-				this.logger.warn(`Persisting default values ${ecc[propertyName]} for engine ${engine}`);
-				await this.writeConfig();
-			} else {
-				throw new Error(`Developer error: no default value set for ${propertyName} of ${engine} engine. Or invalid property call.`)
-			}
-			// Return whatever we came back with.
-			return ecc[propertyName];
+			// If the config doesn't specify the property, use the default value
+			this.logger.trace(`Looking for missing property ${propertyName} of ${engine} in defaults`);
+			ecc = await this.lookupAndUpdateToDefault(engine, ecc, propertyName);
 		}
+
+		return ecc[propertyName];
+	}
+
+	private async lookupAndUpdateToDefault(engine: ENGINE, ecc: EngineConfigContent, propertyName: string): Promise<EngineConfigContent> {
```

Comment:
Mostly the same as existing code, except for line 145 that updates configContent with new engine block.